# P.aphanis transcriptome assembly and annotation pipeline

Documentation of analysis and commands used with P.aphanis as part of Nano Diagnistics PhD.

## Collecting raw RNASeq data

A subfolder for strawberry mildew work was created:
```bash
mkdir P_aphanis
```
All following commands were exectuted from this folder.

Raw sequence data from F.anannassa was copied from long term storage in /archives to the working folder /projects/nano_diagnostics/

```bash
mkdir -p rawdata/P_aphanis/RNAexp1/infected/F
mkdir -p rawdata/P_aphanis/RNAexp1/infected/R
mkdir -p rawdata/P_aphanis/RNAexp1/control/F
mkdir -p rawdata/P_aphanis/RNAexp1/control/R

cp -s /archives/2020_niabemr_general/EUUK2019121203GB-EU-UK-NIAB-EMR-2-RNA-WOBI-H204SC19122617/X204SC19122617-Z01-F001/raw_data/s30008508_1.fq.gz rawdata/P_aphanis/RNAexp1/infected/F

cp -s /archives/2020_niabemr_general/EUUK2019121203GB-EU-UK-NIAB-EMR-2-RNA-WOBI-H204SC19122617/X204SC19122617-Z01-F001/raw_data/s30008508_2.fq.gz rawdata/P_aphanis/RNAexp1/infected/R

cp -s /archives/2020_niabemr_general/EUUK2019121203GB-EU-UK-NIAB-EMR-2-RNA-WOBI-H204SC19122617/X204SC19122617-Z01-F001/raw_data/s30008509_1.fq.gz rawdata/P_aphanis/RNAexp1/control/F

cp -s /archives/2020_niabemr_general/EUUK2019121203GB-EU-UK-NIAB-EMR-2-RNA-WOBI-H204SC19122617/X204SC19122617-Z01-F001/raw_data/s30008509_2.fq.gz rawdata/P_aphanis/RNAexp1/control/R

```

Raw sequence data from F.vesca generated by Jambagi et al. was downloaded.
```bash
wget ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR419/ERR419272/Hawaii0_R1.fq.gz -P rawdata/P_aphanis/Jambagi/ControlHawaii/F
wget ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR419/ERR419272/Hawaii0_R2.fq.gz -P rawdata/P_aphanis/Jambagi/ControlHawaii/R
wget ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR419/ERR419273/Hawaii1dai_R1.fq.gz -P rawdata/P_aphanis/Jambagi/1DAIHawaii/F
wget ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR419/ERR419273/Hawaii1dai_R2.fq.gz -P rawdata/P_aphanis/Jambagi/1DAIHawaii/R
wget ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR419/ERR419274/Hawaii8dai_R1.fq.gz -P rawdata/P_aphanis/Jambagi/8DAIHawaii/F
wget ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR419/ERR419274/Hawaii8dai_R2.fq.gz -P rawdata/P_aphanis/Jambagi/8DAIHawaii/R
wget ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR424/ERR424891/YellowWonder0_R1.fq.gz -P rawdata/P_aphanis/Jambagi/ControlYellowWonder/F
wget ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR424/ERR424891/YellowWonder0_R2.fq.gz -P rawdata/P_aphanis/Jambagi/ControlYellowWonder/R
wget ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR424/ERR424892/YellowWonder1dai_R1.fq.gz -P rawdata/P_aphanis/Jambagi/1DAIYellowWonder/F
wget ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR424/ERR424892/YellowWonder1dai_R2.fq.gz -P rawdata/P_aphanis/Jambagi/1DAIYellowWonder/R
wget ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR424/ERR424893/YellowWonder8dai_R1.fq.gz -P rawdata/P_aphanis/Jambagi/8DAIYellowWonder/F
wget ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR424/ERR424893/YellowWonder8dai_R2.fq.gz -P rawdata/P_aphanis/Jambagi/8DAIYellowWonder/R
```

## Quality controlling RNASeq data

Fastqc was installed

A new fastqc wrapper script was created based upon script from A.Armitage

The raw sequence reads were subjected to a quality control check using FastQC.
```bash
for RawData in $(ls rawdata/*/*/*/*/*.fq.gz); do
echo $RawData
ProgDir=~/git_repos/tools/seq_tools/dna_qc
OutDir=$(dirname $RawData)
sbatch $ProgDir/srun_fastqc.sh $RawData $OutDir
done
```

A new trimmomatic wrapper script was created based upon script from A.Armitage 

Raw sequence reads were trimmed using trimmomatic

```bash
for ReadDir in $(ls -d rawdata/*/*/*); do
 Fread=$(ls $ReadDir/F/*.fq.gz)
 Rread=$(ls $ReadDir/R/*.fq.gz)
ls $Fread
ls $Rread
Adapters=~/git_repos/tools/seq_tools/ncbi_adapters.fa
ProgDir=~/git_repos/tools/seq_tools/dna_qc
OutDir=$(echo $ReadDir|sed 's@rawdata@dna_qc@g')
Prefix=$(echo $ReadDir|cut -f2,3,4,5 -d '/' --output-delimiter '-')
sbatch $ProgDir/srun_trimmomatic.sh $Fread $Rread $Adapters $OutDir $Prefix
done
```
The trimmed reads were subjected to a quality control check using FastQC.
```bash
for RawData in $(ls dna_qc/*/*/*/*/*trim.fq.gz); do
echo $RawData
ProgDir=~/git_repos/tools/seq_tools/dna_qc
OutDir=$(dirname $RawData)
sbatch $ProgDir/srun_fastqc.sh $RawData $OutDir
done
```
The number of reads were counted
```bash
cat dna_qc/P_aphanis/RNAexp1/infected/F/P_aphanis-RNAexp1-infected_F_trim.fq.gz | gunzip -cf | wc -l
#88,436,408

cat dna_qc/P_aphanis/RNAexp1/infected/R/P_aphanis-RNAexp1-infected_R_trim.fq.gz | gunzip -cf | wc -l
#88,436,408

cat dna_qc/P_aphanis/RNAexp1/control/R/P_aphanis-RNAexp1-control_R_trim.fq.gz | gunzip -cf | wc -l
#79,689,604

cat dna_qc/P_aphanis/RNAexp1/control/F/P_aphanis-RNAexp1-control_F_trim.fq.gz | gunzip -cf | wc -l
#79,689,604
```

## Alignment

The most up to date version of the camarosa genome was downloaded as well as annotation information.

```bash
srun -p long --pty bash

cd /projects/nano_diagnostics/rawdata/F_ananassa/camarosa/genome

wget ftp://ftp.bioinfo.wsu.edu/species/Fragia_x_ananassa/Fragaria_x_ananassa_Camarosa_Genome_v1.0.a1/assembly/F_ana_Camarosa_6-28-1fasta.gz

gunzip F_ana_Camarosa_6-28-17.fasta.gz

mkdir -p /projects/nano_diagnostics/rawdata/F_ananassa/camarosa/annotations

cd /projects/nano_diagnostics/rawdata/F_ananassa/camarosa/annotations

wget ftp://ftp.bioinfo.wsu.edu/species/Fragaria_x_ananassa/Fragaria_x_ananassa_Camarosa_Genome_v1.0.a1/genes/Fxa_v1.2_makerStandard_MakerGenes_woTposases.gff.gz

gunzip Fxa_v1.2_makerStandard_MakerGenes_woTposases.gff.gz
```
### STAR

The programme Star was installed for fast RNAseq data alignment

A new STAR wrapper script was created based upon script from A.Armitage 

The trimmed reads were aligned to genome of F.ananassa 'camarosa' using STAR, annotations were used to extract splice junctions and improve the accuracy of mapping.

The STAR wrapper script was changed to save unaligned read information and to accept .gff annotation files using the following settings: --sjdbGTFtagExonParentTranscript Parent, --sjdbOverhang 149, --outReadsUnmapped Fastx, --outSAMunmapped Within

```bash
for ReadDir in $(ls -d dna_qc/*/*/*); do
 Fread=$(ls $ReadDir/F/*trim.fq.gz)
 Rread=$(ls $ReadDir/R/*trim.fq.gz)
ls $Fread
ls $Rread
InGenome=rawdata/F_ananassa/camarosa/genome/F_ana_Camarosa_6-28-17.fasta 
InGff=rawdata/F_ananassa/camarosa/annotations/Fxa_v1.2_makerStandard_MakerGenes_woTposases.gff
ProgDir=~/git_repos/tools/seq_tools/RNAseq
OutDir=$(echo $ReadDir|sed 's@dna_qc@alignment/STAR@g')
sbatch $ProgDir/ssub_star.sh $InGenome $Fread $Rread $OutDir $InGff
done
```
## Transcriptome Assembly

### Trinity

The first method tried to generate a denovo P.aphanis transcriptome was using the assembler trinity.

Trinity was installed.

Trinity runs with sample data demonstrated that using .fq files was resulting in erros. Quality controlled reads were therefore converted from .fq to .fa
```bash
#Files were uncompressed
for ReadDir in $(ls -d dna_qc/P_aphanis/RNAexp1/*); do
 Fread=$(ls $ReadDir/F/*trim.fq.gz)
 Rread=$(ls $ReadDir/R/*trim.fq.gz)
ls $Fread
ls $Rread
gunzip -c $Fread > $(echo $Fread|sed 's@fq.gz@fastq@g')
gunzip -c $Rread > $(echo $Rread|sed 's@fq.gz@fastq@g')
done

#Files were converted to fa format
for ReadDir in $(ls -d dna_qc/P_aphanis/RNAexp1/*); do
 Fread=$(ls $ReadDir/F/*trim.fastq)
 Rread=$(ls $ReadDir/R/*trim.fastq)
ls $Fread
ls $Rread
sed -n '1~4s/^@/>/p;2~4p' $Fread > $(echo $Fread|sed 's@fastq@fa@g')
sed -n '1~4s/^@/>/p;2~4p' $Rread > $(echo $Rread|sed 's@fastq@fa@g')
done
```
Reads that did not align to the strawberry genome were also converted to .fa
```bash
#for unaligned reads
for Unaligned in $(ls -d alignment/STAR/P_aphanis/RNAexp1/*/); do
 Fread=$(ls $Unaligned*.mate1)
 Rread=$(ls $Unaligned*.mate2)
ls $Fread
ls $Rread
sed -n '1~4s/^@/>/p;2~4p' $Fread > $(echo $Fread|sed 's@mate1@.F.fa@g')
sed -n '1~4s/^@/>/p;2~4p' $Rread > $(echo $Rread|sed 's@mate2@.R.fa@g')
done
```
De novo transcriptome assembly was performed for combined quality controlled reads and unaligned reads. Trinity can be memory intensive, the settings used for these runs were as follows; #SBATCH --partition=long, #SBATCH --nodes=1, #SBATCH --ntasks=1, #SBATCH --cpus-per-task=10, #SBATCH --mem=450G,  --CPU 10 \,  --max_memory 350G \,  --seqType fa \,  --SS_lib_type FR \,  --left $InReadF \,  --right $InReadR 

The following were not used here but may help reduce memory requirements; --min_kmer_cov 2, --bflyCalculateCPU
More details at; http://trinityrnaseq.github.io/performance/mem.html and https://github.com/trinityrnaseq/trinityrnaseq/wiki
```bash
#a trinity instance of bioconda was opened
screen -S conda_trinity
conda activate trinity
#Trinity de novo transcriptome assembly was run for all quality controlled reads
for ReadDir in $(ls -d dna_qc/P_aphanis/RNAexp1/*); do
 Fread=$(ls $ReadDir/F/*trim.fa)
 Rread=$(ls $ReadDir/R/*trim.fa)
ls $Fread
ls $Rread
ProgDir=~/git_repos/tools/seq_tools/RNAseq
OutDir=$(echo $ReadDir|sed 's@dna_qc@assembly/trancriptome/trinity@g')
sbatch $ProgDir/ssub_trinity.sh $Fread $Rread $OutDir 
done
#309671 - control (medium partion)
#309674 - infected (long partion)

#Trinity de novo transcriptome assembly was run for unnmapped reads
for ReadDir in $(ls -d alignment/STAR/P_aphanis/RNAexp1/*); do
 Fread=$(ls $ReadDir/*.F.fa)
 Rread=$(ls $ReadDir/*.R.fa)
ls $Fread
ls $Rread
ProgDir=~/git_repos/tools/seq_tools/RNAseq
OutDir=$(echo $ReadDir/unaligned|sed 's@alignment/STAR@assembly/trancriptome/trinity@g')
sbatch $ProgDir/ssub_trinity.sh $Fread $Rread $OutDir 
done
#309675 - control (himem)
#309676 - infected (himem)
```

# From here on pipeline is a work in progress

The software rnaQUAST for quality evaluation and assessment of de novo transcriptome assemblies was installed along with dependencies; gffutils, matplotlib, joblib, GMAP, BLASTN.

When a reference genome is unavailable it is recommended to run rnaQUAST with BUSCO and GeneMarkS-T options. This required the additional installation of depedancies; GeneMarkS-T, BUSCO, tblastn, HMMER and transeq.











Quast was used to assess the quality of genome assembly:
```bash
  ProgDir=/home/heavet/git_repos/tools/seq_tools/assemblers/assembly_qc/quast
    for Assembly in $(ls assembly/spades/*/*/filtered_contigs/contigs_min_500bp.fasta); do
    Strain=$(echo $Assembly | rev | cut -f3 -d '/' | rev)
    Organism=$(echo $Assembly | rev | cut -f4 -d '/' | rev)  
    OutDir=assembly/spades/$Organism/$Strain/filtered_contigs
    qsub $ProgDir/sub_quast.sh $Assembly $OutDir
  done
```
The results of quast were desplayed:
```bash
  for Assembly in $(ls assembly/spades/*/*/filtered_contigs/report.txt); do
    Strain=$(echo $Assembly | rev | cut -f3 -d '/' | rev);
    Organism=$(echo $Assembly | rev | cut -f4 -d '/' | rev);
    echo;
    echo $Organism;
    echo $Strain;
    cat $Assembly;
  done > assembly/quast_results.txt
```

The podosphaera xanthii transcriptome was downloaded - assembled using MIRA3 & euler by D.Corsia
```bash
mkdir -p rawdata/P_xanthii/D_CorsiaExp/transcriptome/

srun -p long --pty bash

cd rawdata/P_xanthii/D_CorsiaExp/transcriptome/

wget https://sra-download.ncbi.nlm.nih.gov/traces/wgs03/wgs_aux/GE/UO/GEUO01/GEUO01.1.gbff.gz

wget https://sra-download.ncbi.nlm.nih.gov/traces/wgs03/wgs_aux/GE/UO/GEUO01/GEUO01.1.fsa_nt.gz

gunzip GEUO01.1.fsa_nt.gz # fasta file

gunzip GEUO01.1.gbff.gz #Genebank file
```
The podosphaera pannosa transcriptome was downloaded - assembled using trinity by N.Fonseca
```bash
mkdir -p rawdata/P_pannosa/N_FonsecaExp/transcriptome/

srun -p long --pty bash

cd rawdata/P_pannosa/N_FonsecaExp/transcriptome/

wget https://sra-download.ncbi.nlm.nih.gov/traces/wgs01/wgs_aux/GH/DE/GHDE01/GHDE01.1.fsa_nt.gz

wget https://sra-download.ncbi.nlm.nih.gov/traces/wgs01/wgs_aux/GH/DE/GHDE01/GHDE01.1.gbff.gz

gunzip GHDE01.1.fsa_nt.gz # fasta file

gunzip GHDE01.1.gbff.gz #Genebank file
```





A partial genome assembly for P.aphanis genertaed by H.Cockerton was collected.

```bash
mkdir -p /projects/nano_diagnostics/assembly/genome/spades/P_aphanis/H_Cockerton/

cp -s -r /projects/oldhome/groups/harrisonlab/project_files/podosphaera/assembly/spades/P.aphanis/ /projects/nano_diagnostics/assembly/genome/spades/P_aphanis/H_Cockerton/

ls /projects/nano_diagnostics/assembly/genome/spades/P_aphanis/H_Cockerton/C1_no_strawberry/deconseq_appended/contigs_min_500bp_renamed.fasta
```
STAR was used to align trimmed reads that did not align to F.ananassa to the P.aphanis genome

```bash
for ReadDir in $(ls -d alignment/STAR/P_aphanis/RNAexp1/*); do
 Fread=$(ls $ReadDir/*.F.fa)
 Rread=$(ls $ReadDir/*.R.fa)
ls $Fread
ls $Rread
InGenome=assembly/genome/spades/P_aphanis/H_Cockerton/C1_no_strawberry/deconseq_appended/contigs_min_500bp_renamed.fasta 
#InGff=
ProgDir=~/git_repos/tools/seq_tools/RNAseq
OutDir=$(echo $ReadDir/P_aphanis)
sbatch $ProgDir/ssub_star.sh $InGenome $Fread $Rread $OutDir
done
#318140
#318141
```